### PR TITLE
Create update-microcode

### PR DESCRIPTION
**Pull request by rizalmart**
This script updates the CPU microcode easily

### CODE REVIEW

File: woof-code/rootfs-skeleton/usr/update-microcode [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+#manually update cpu microcode
+#written by mistfire
+
+ if [ "$(uname -m | grep -E '^x86_64|^amd64|^i.86')" == "" ]; then
+  echo "Not x86/x64 CPU"
+  exit
+ fi	
+
+ if [ "$(which lscpu)" == "" ]; then
```
Comment (01micko) [inline woof-code/rootfs-skeleton/usr/update-microcode]: No need for `lscpu`, just scan `/proc/cpuinfo`

File: woof-code/rootfs-skeleton/usr/update-microcode [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ fi
+ 
+ if [ "$(lscpu | grep "Hypervisor")" != "" ]; then
+  echo "Not a physical CPU"
+  exit
+ fi
+
+  xCPU=""
+  
+  [ "$(lscpu | grep "Intel")" != "" ] && xCPU='intel'
```
Comment (01micko) [inline woof-code/rootfs-skeleton/usr/update-microcode]: `grep -qm1 "Intel" /proc/cpuinfo && xCPU='intel' # tiny bit faster too`

**Review by 01micko — CHANGES_REQUESTED** [review]

**Comment by 01micko** [comment]
The AMD microcode is in the kernel.org firware repo, but the intel microcode is at the intel github repo. https://github.com/intel/Intel-Linux-Processor-Microcode-Data-Files

These *could* be added to the `fdrv` as part of `kernel-kit` :thinking: 

Alternatively we could add firmware to the `initrd.gz` in the `3builddistro` stage and have early microcode loading (dunno what is 'micro' in this code, files are rather large! :fearful: ) see https://www.kernel.org/doc/Documentation/x86/microcode.txt

**Comment by 01micko** [comment]
Well, early loading works too, with the two initrds concatenated.. amazing. (as in the linked document)
```
$ dmesg | grep -i microcode
[    0.000000] microcode: microcode updated early to revision 0xea, date = 2021-01-25
[    0.128397] SRBDS: Mitigation: Microcode
[    1.449204] microcode: sig=0x406e3, pf=0x80, revision=0xea
[    1.449231] microcode: Microcode Update Driver: v2.2.
```

**Comment by 01micko** [comment]
@rizalmart please read @jamesbond3142 's respone to #2344 and give your opinion. Thanks.

File: woof-code/rootfs-skeleton/usr/update-microcode [hunk 3]
```diff
@@ -24,10 +24,10 @@
 
   xCPU=""
   
-  [ "$(lscpu | grep "Intel")" != "" ] && xCPU='intel' 
+  [ "$(cat /proc/cpuinfo | grep -m 1 "Intel")" != "" ] && xCPU='intel' 
   
   if [ "$xCPU" == "" ]; then
-   [ "$(lscpu | grep "AMD")" != "" ] && xCPU='amd'   
+   [ "$(cat /proc/cpuinfo | grep -m 1 "AMD")" != "" ] && xCPU='amd'
```
Comment (dimkr) [inline woof-code/rootfs-skeleton/usr/update-microcode]: You can even avoid this `cat`: ` grep -m 1 "AMD" /proc/cpuinfo`

Comment (01micko) [reply woof-code/rootfs-skeleton/usr/update-microcode]: Of course.

**Pull request closed without merging**